### PR TITLE
use service loader with gziphandler

### LINKED ISSUE #8769: Introduce CompressionHandler to support compression from gzip, brotli, and zstandard
**Jetty version(s)**
Jetty 12

**Enhancement Description**
The current `GzipHandler` does a good job at handling `Content-Encoding: gzip` (and `Accept-Encoding: gzip`), but we shouldn't hardcode only gzip support.

Can we refactor the existing `GzipHandler` to be more generic, something like `DynamicCompressionHandler` and have the Gzip support (courtesy of the existing `GzipRequest` object) just be loaded via a ServiceLoader?

This way we can encourage other compression algorithms under the same `Accept-Encoding` / `Content-Encoding` / already compressed rules that the existing codebase handles for us?

Other compression algorithms that have been requested in our issue tracker.

* #2553 - [RFC 7932](https://www.rfc-editor.org/rfc/rfc7932) - Brotli Compressed Data Format 
* #6600 - [RFC 8878](https://www.rfc-editor.org/rfc/rfc8878) - Zstandard Compression and the `application/zstd` Media Type

(FYI, `curl` supports both of these compression types now)

### PR DESCRIPTION
**Pull request by jimexist**
- closes #8769

### CODE REVIEW

**Comment by joakime** [comment]
There's no test cases show how this would be used.

Can you create a ExampleCompression whatever in the src/test/java and src/test/resources/META-INF/services/ to show it working?  The compression doesn't have to do actual compression, heck, make it just ROT13 for purposes of the test. (that way we can see it performing the transformation and back again)

File: documentation/jetty-documentation/src/main/java/org/eclipse/jetty/docs/programming/server/http/HTTPServerDocs.java [hunk 1]
```diff
@@ -876,7 +878,7 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
         shopContext.setHandler(new ShopHandler());
 
         // You want to gzip the shop web application only.
-        GzipHandler shopGzipHandler = new GzipHandler();
+        DynamicCompressionHandler shopGzipHandler = ServiceLoader.load(DynamicCompressionHandler.class).iterator().next();
```
Comment (janbartel) [inline documentation/jetty-documentation/src/main/java/org/eclipse/jetty/docs/programming/server/http/HTTPServerDocs.java:881]: I can't see the benefit of using the `ServiceLoader` this way. A user can just extend `GzipHandler` and then call `ServletContextHandler.setGzipHandler`.

Can you explain the problem you're trying to solve a bit more?

**Review by janbartel — CHANGES_REQUESTED** [review]

**Comment by gregw** [comment]
Sorry we have not acted on this PR for while. Busy month.
Will tune in next month.

File: documentation/jetty-documentation/src/main/java/org/eclipse/jetty/docs/programming/server/http/HTTPServerDocs.java (hunk 1 above)
Comment (jimexist) [reply documentation/jetty-documentation/src/main/java/org/eclipse/jetty/docs/programming/server/http/HTTPServerDocs.java:881]: specific approach inspired by original issue https://github.com/eclipse/jetty.project/issues/8769 but personally i think it's okay to have it either way as long as interface is consistent

**Pull request closed without merging**